File: social-crawler/weibo_scraper.mjs
import playwright from 'playwright';

const brands = [
  { name: '瑞幸咖啡', uid: '5308260857' },
  { name: '库迪咖啡', uid: '7497928702' },
  { name: '古茗茶饮', uid: '7480804260' },
  { name: '茉莉奶白', uid: '7541451616' },
  { name: '霸王茶姬', uid: '7453081605' },
  { name: '喜茶', uid: '612941' },
  { name: '乐乐茶', uid: '6212294155' },
  { name: 'T9tea', uid: '7609260234' },
  { name: '奈雪的茶', uid: '6615096941' },
  { name: '一只酸奶牛', uid: '740685496' },
  { name: '益禾堂', uid: '7468036877' },
  { name: 'CoCo都可', uid: '164208115' },
  { name: '书亦烧仙草', uid: '6617749221' },
  { name: '悸动烧仙草', uid: '7469168385' },
  { name: '沪上阿姨', uid: '7459931033' },
  { name: '茶百道', uid: '6615096942' },
  { name: '七分甜', uid: '7458273805' },
];

async function scrapeBrand(browser, brand) {
  const context = browser.contexts()[0];
  const page = await context.newPage();
  
  try {
    await page.goto(`https://weibo.com/u/${brand.uid}`, { 
      waitUntil: 'networkidle',
      timeout: 20000 
    });
    
    // Wait for content to load
    await page.waitForTimeout(3000);
    
    // Try to extract posts from the page
    const posts = await page.evaluate(() => {
      const items = document.querySelectorAll('.woo-panel-main');
      const results = [];
      
      // Look for Weibo post items
      const postItems = document.querySelectorAll('[action-type="feed_list_item"], .vue-recycle-scroller__item-view');
      
      document.querySelectorAll('.WB_feed .WB_cardwrap, .WB_feed_type').forEach(el => {
        const contentEl = el.querySelector('.WB_text, .feed_content');
        const timeEl = el.querySelector('.WB_from, .feed_from');
        const likeEl = el.querySelector('.WB_row_line li:nth-child(3) span, .item_agree');
        const cmtEl = el.querySelector('.WB_row_line li:nth-child(2) a, .item评论');
        const repostsEl = el.querySelector('.WB_row_line li:nth-child(1) a, .item转发');
        
        if (contentEl) {
          results.push({
            content: contentEl.innerText?.trim() || '',
            time: timeEl?.innerText?.trim() || '',
            likes: likeEl?.innerText?.trim() || '',
            comments: cmtEl?.innerText?.trim() || '',
            reposts: repostsEl?.innerText?.trim() || ''
          });
        }
      });
      
      return results;
    });
    
    await page.close();
    return { name: brand.name, uid: brand.uid, posts, error: null };
  } catch (e) {
    await page.close();
    return { name: brand.name, uid: brand.uid, posts: [], error: e.message };
  }
}

async function main() {
  console.log('Connecting to Chrome via CDP...');
  const browser = await playwright.chromium.connectOverCDP('http://localhost:9333');
  
  console.log('Scraping Weibo for tea brands...');
  const results = [];
  
  for (const brand of brands) {
    console.log(`Scraping ${brand.name} (${brand.uid})...`);
    const result = await scrapeBrand(browser, brand);
    results.push(result);
    console.log(`  -> Got ${result.posts.length} posts, error: ${result.error || 'none'}`);
    
    // Small delay between requests
    await new Promise(r => setTimeout(r, 1500));
  }
  
  console.log('\n--- SCRAPED DATA ---');
  console.log(JSON.stringify(results, null, 2));
  
  await browser.close();
}

main().catch(console.error);

File: social-crawler/weibo_scraper.py
#!/usr/bin/env python3
"""
Scrape Weibo brand content using OpenClaw's managed Chrome (port 18800).
Each brand is loaded fresh, JS extracts posts, then we move to next brand.
"""
import json
import subprocess
import time
import os

BRANDS = [
    ("瑞幸咖啡", "https://weibo.com/u/6349791448"),
    ("库迪", "https://weibo.com/u/7791266545"),
    ("古茗", "https://weibo.com/u/2809775704"),
    ("茉莉奶白", "https://weibo.com/u/7577524421"),
    ("霸王茶姬", "https://weibo.com/u/5652018762"),
    ("喜茶", "https://weibo.com/u/2804387887"),
    ("星巴克", "https://weibo.com/starbucks"),
    ("茶百道", "https://weibo.com/u/6502206666"),
    ("奈雪的茶", "https://weibo.com/u/5884674413"),
    ("CoCo", "https://weibo.com/u/2030619861"),
    ("爷爷不泡茶", "https://weibo.com/u/7769072120"),
    ("沪上阿姨", "https://weibo.com/u/3921865344"),
    ("乐乐茶", "https://weibo.com/u/6253473981"),
    ("皮爷咖啡", "https://weibo.com/u/6360528436"),
    ("M Stand", "https://weibo.com/u/6345199298"),
    ("Manner", "https://weibo.com/u/6808111794"),
    ("茉酸奶", "https://weibo.com/u/5188894132"),
    ("树夏酸奶", "https://weibo.com/u/7144806571"),
]

EXTRACTOR = """() => {
  const delay = ms => new Promise(r => setTimeout(r, ms));
  const scroll = async () => {
    for (let i = 0; i < 5; i++) {
      window.scrollBy(0, 500);
      await delay(300);
    }
    window.scrollTo(0, 0);
  };
  await scroll();
  await delay(1000);
  
  // Find all post containers - try multiple selectors
  let posts = [];
  const containers = document.querySelectorAll('[class*="_content_"], [class*="item_"]');
  
  // Try to get all text content blocks
  const allContent = [];
  const seen = new Set();
  
  // Look for the main feed area
  const feedArea = document.querySelector('[class*="woo"]') || document.body;
  const paragraphs = feedArea.querySelectorAll('p, div[style], span');
  
  // Try scrolling approach - get all text visible after scrolling
  const results = {
    title: document.title,
    url: window.location.href,
    bodyText: document.body.innerText.slice(0, 8000),
    html: document.body.innerHTML.slice(0, 50000)
  };
  
  return JSON.stringify(results);
}"""

def get_tab_id():
    """Get the tab ID for the OpenClaw browser."""
    result = subprocess.run(
        ["curl", "-s", "http://127.0.0.1:18800/json"],
        capture_output=True, text=True
    )
    if result.returncode != 0:
        return None
    try:
        data = json.loads(result.stdout)
        for tab in data:
            if tab.get("type") == "page" and "weibo" in tab.get("url", ""):
                return tab["id"]
        if data:
            return data[0]["id"]
    except:
        pass
    return None

def navigate_and_extract(tab_id, url, brand_name):
    """Navigate to URL and extract content via CDP."""
    # Use browser tool to navigate
    import requests
    
    # First navigate
    ws_url = f"ws://127.0.0.1:18800/devtools/page/{tab_id}"
    from websocket import create_connection
    
    def send_cmd(ws, method, params=None):
        ws.send(json.dumps({"id": 1, "method": method, "params": params or {}}))
        return json.loads(ws.recv())
    
    try:
        ws = create_connection(ws_url)
        # Navigate
        send_cmd(ws, "Page.navigate", {"url": url})
        time.sleep(5)  # Wait for page load
        
        # Scroll and get content
        send_cmd(ws, "Runtime.evaluate", {
            "expression": """
            (async () => {
              for(let i=0; i<5; i++) { window.scrollBy(0,400); await new Promise(r=>setTimeout(r,400)); }
              window.scrollTo(0,0);
              await new Promise(r=>setTimeout(r,1000));
              return document.body.innerText.slice(0, 8000);
            })()
            """,
            "awaitPromise": True
        })
        
        resp = send_cmd(ws, "Runtime.evaluate", {
            "expression": "document.body.innerText.slice(0, 8000)"
        })
        ws.close()
        
        text = resp.get("result", {}).get("value", "") if resp else ""
        return text
    except Exception as e:
        return f"Error: {e}"

# For now, use the browser tool API
def main():
    # Get tab ID
    tab_id = get_tab_id()
    print(f"Tab ID: {tab_id}")
    
    for brand_name, url in BRANDS:
        print(f"\\n=== Scraping {brand_name}: {url} ===")
        text = navigate_and_extract(tab_id, url, brand_name)
        print(text[:500])
        time.sleep(2)

if __name__ == "__main__":
    main()
